### PR TITLE
add GPU support for Azure provider

### LINKED ISSUE #6601: Add GPU support for Azure provider

### PR DESCRIPTION
**Pull request by zreigz**
**What this PR does / why we need it**: add GPU support for Azure provider

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #6601

```release-note
add GPU support for Azure provider
```

### CODE REVIEW

File: pkg/handler/common/provider/azure.go [hunk 1]
```diff
@@ -299,6 +299,14 @@ func isValidVM(sku compute.ResourceSku, location string) bool {
 }
 
 func AzureSize(ctx context.Context, quota kubermaticv1.MachineDeploymentVMResourceQuota, subscriptionID, clientID, clientSecret, tenantID, location string) (apiv1.AzureSizeList, error) {
+	// https://docs.microsoft.com/en-us/azure/virtual-machines/sizes-gpu
+	gpuInstanceFamilies := map[string]int32{"Standard_NC6": 1, "Standard_NC12": 2, "Standard_NC24": 4, "Standard_NC24r": 4,
+		"Standard_NC6s_v2": 1, "Standard_NC12s_v2": 2, "Standard_NC24s_v2": 4, "Standard_NC24rs_v2": 4, "Standard_NC6s_v3": 1,
+		"Standard_NC12s_v3": 2, "Standard_NC24s_v3": 4, "Standard_NC24rs_v3": 4, "Standard_NC4as_T4_v3": 1, "Standard_NC8as_T4_v3": 1,
+		"Standard_NC16as_T4_v3": 1, "Standard_NC64as_T4_v3": 4, "Standard_ND6s": 1, "Standard_ND12s": 2, "Standard_ND24s": 4, "Standard_ND24rs": 4,
+		"Standard_ND40rs_v2": 8, "Standardowa_NV6": 1, "Standardowa_NV12": 2, "Standardowa_NV24": 4, "Standard_NV12s_v3": 1, "Standard_NV24s_v3": 2, "Standard_NV48s_v3": 4,
```
Comment (lsviben) [inline pkg/handler/common/provider/azure.go]: `"Standardowa_NV6": 1, "Standardowa_NV12": 2, "Standardowa_NV24": 4` these should be "Standard_XXX" right?

File: pkg/handler/common/provider/azure.go [hunk 2]
```diff
@@ -299,6 +299,14 @@ func isValidVM(sku compute.ResourceSku, location string) bool {
 }
 
 func AzureSize(ctx context.Context, quota kubermaticv1.MachineDeploymentVMResourceQuota, subscriptionID, clientID, clientSecret, tenantID, location string) (apiv1.AzureSizeList, error) {
+	// https://docs.microsoft.com/en-us/azure/virtual-machines/sizes-gpu
+	gpuInstanceFamilies := map[string]int32{"Standard_NC6": 1, "Standard_NC12": 2, "Standard_NC24": 4, "Standard_NC24r": 4,
```
Comment (lsviben) [inline pkg/handler/common/provider/azure.go]: i would move this a package var so the map is not initialized every time AzureSize is called.

**Review by lsviben — CHANGES_REQUESTED** [review]
2 small comments. 

Too bad that there is no azure API to get the GPU sizes, now we will need to remember to update this GPU sizes map every once in a while or until the client gets the GPU amount possibility.

File: pkg/handler/common/provider/azure.go (hunk 1 above)
Comment (zreigz) [reply pkg/handler/common/provider/azure.go]: done

File: pkg/handler/common/provider/azure.go (hunk 2 above)
Comment (zreigz) [reply pkg/handler/common/provider/azure.go]: done

**Pull request merged**